lighthouse-core/report/v2/renderer/report-features.js

```diff
@@ -0,0 +1,326 @@
+    if (this._copyAttempt) {
+      // We want to write our own data to the clipboard, not the user's text selection.
+      e.preventDefault();
+      e.clipboardData.setData('text/plain', JSON.stringify(this.json, null, 2));
+      this.logger.log('Report JSON copied to clipboard');
+    }
+
+    this._copyAttempt = false;
+  }
+
+  /**
+   * Copies the report JSON to the clipboard (if supported by the browser).
+   */
+  onCopyButtonClick() {
+    this._fireEventOn('lh-analytics', this._document, {
+      cmd: 'send',
+      fields: {hitType: 'event', eventCategory: 'report', eventAction: 'copy'}
+    });
+
+    try {
+      if (this._document.queryCommandSupported('copy')) {
+        this._copyAttempt = true;
+
+        // Note: In Safari 10.0.1, execCommand('copy') returns true if there's
+        // a valid text selection on the page. See http://caniuse.com/#feat=clipboard.
+        if (!this._document.execCommand('copy')) {
+          this._copyAttempt = false; // Prevent event handler from seeing this as a copy attempt.
+          this.logger.warn('Your browser does not support copy to clipboard.');
+        }
+      }
+    } catch (/** @type {!Error} */ e) {
+      this._copyAttempt = false;
+      this.logger.log(e.message);
+    }
+  }
+
+  closeExportDropdown() {
+    this.exportButton.classList.remove('active');
+  }
+
+  /**
+   * Click handler for export button.
+   * @param {!Event} e
+   */
+  onExportButtonClick(e) {
+    e.preventDefault();
+    const el = /** @type {!Element} */ (e.target);
+    el.classList.toggle('active');
+    this._document.addEventListener('keydown', this.onKeyDown);
+  }
+
+  /**
+   * Resets the state of page before capturing the page for export.
+   * When the user opens the exported HTML page, certain UI elements should
+   * be in their open state (not opened) and the templates should be refreshly
+   * stamped.
+   */
+  _resetUIState() {
+    this.logger.hide();
+    this.closeExportDropdown();
+    this._dom.findAll('template[data-stamped]', this._document).forEach(t => {
+      t.removeAttribute('data-stamped');
+    });
+  }
+
+  /**
+   * Handler for "export as" button.
+   * @param {!Event} e
+   */
+  onExport(e) {
+    e.preventDefault();
+
+    const el = /** @type {!Element} */ (e.target);
+
+    if (!el.hasAttribute('data-action')) {
+      return;
+    }
+
+    switch (el.getAttribute('data-action')) {
+      case 'copy':
+        this.onCopyButtonClick();
+        break;
+      case 'open-viewer':
+        this.sendJSONReport();
+        break;
+      case 'print':
+        this.expandAllDetails();
+        self.print();
+        break;
+      case 'save-json': {
+        const jsonStr = JSON.stringify(this.json, null, 2);
+        this._saveFile(new Blob([jsonStr], {type: 'application/json'}));
+        break;
+      }
+      case 'save-html': {
+        this._resetUIState();
+
+        const htmlStr = this._document.documentElement.outerHTML;
```

Comment:
Unfortunately, we need the whole page (outer shell, templates, styles, etc.) when exporting to html. [Same as before.](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/report/scripts/lighthouse-report.js#L147).